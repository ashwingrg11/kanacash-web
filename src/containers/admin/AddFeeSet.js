import React, { Component } from "react";
import { Row, Col, Card, Button } from "react-bootstrap";
import COLORS from "../../assets/Color/Color";
import { Dropdown } from "../../components/CustomDropdown";
import AdminDropdown from "../../components/CustomDropdown/AdminDropdown";
import { InputField } from "../../components/CustomInput";
import plusRound from "../../assets/images/admin/plus-round.svg";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { isEmpty } from "../../utils/isEmpty";
import ButtonSpinner from "../../components/ButtonSpinner/ButtonSpinner";
import * as api from "../../services/axios/adminApi";
import PropTypes from "prop-types";
import { getFeeStructure } from "../../store/actions/adminAction";
import deleteImg from "../../assets/images/icons/delete.png";
import ROUTES from "../../assets/Routes/Routes";
import { getActiveAccountSideBar } from "../../store/actions/senderDetail";

class AddFeeSet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sourceDestination: [],
      paymentPayout: [
        {
          paymentMethod: "BANK",
          payoutMethod: "BANK_DEPOSIT",
        },
        {
          paymentMethod: "CARD",
          payoutMethod: "BANK_DEPOSIT",
        },
      ],
      currency: [{ threeCharCode: "USD" }, { threeCharCode: "NGA" }],
      selectedSourceDestination: {},
      selectedPaymentPayout: {},
      isSubmitting: false,
      selectedCurrency: {},
      feeSet: [
        { minAmount: "", maxAmount: "", flatFee: "", percentageFee: "" },
      ],
      edited: false,
      showAlert: false,
      message: "",
      variant: "",
      errorMessage: "",
      errorKey: "",
      errorIndex: "",
    };
  }
  componentDidMount = () => {
    if (this.props.location.data) {
      let data = this.props.location.data;
      this.setState({
        selectedSourceDestination: {
          destName: "nigeria",
          destThreeCharCode: data.destinationCountry,
          destTwoCharCode: "ng",
          destFlagUrl: `https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.2.1/flags/4x3/ng.svg`,
          sourceName: "United Stage",
          sourceThreeCharCode: "USA",
          sourceTwoCharCode: "US",
          sourceFlagUrl:
            "https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.2.1/flags/4x3/us.svg",
        },
        selectedPaymentPayout: {
          paymentMethod: data.paymentMethod,
          payoutMethod: data.payoutMethod,
        },
        isSubmitting: false,
        selectedCurrency: { threeCharCode: data.currency },
        feeSet: data.feeRanges,
        edited: true,
      });
    } else {
    }
    this.getSourceDestination(this.props.countries);
  };

  getSourceDestination = (data) => {
    let sourceDestination = [];
    data.map((item) => {
      if (item.three_char_code !== "USA") {
        sourceDestination.push({
          destName: item.name,
          destThreeCharCode: item.three_char_code,
          destTwoCharCode: item.two_char_code,
          destFlagUrl: `https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.2.1/flags/4x3/${item.two_char_code.toLowerCase()}.svg`,
          sourceName: "United Stage",
          sourceThreeCharCode: "USA",
          sourceTwoCharCode: "US",
          sourceFlagUrl:
            "https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.2.1/flags/4x3/us.svg",
        });
      }
      return true;
    });
    this.setState({ sourceDestination });
  };

  componentDidUpdate = (prevProps) => {
    if (prevProps.countries !== this.props.countries) {
      this.getSourceDestination(this.props.countries);
    }
  };

  handleChange = (values) => {
    let { name, value } = values;
    this.setState({ [name]: value });
  };
  addFeeStructure = () => {
    this.error = false;
    this.maxAmount = 0;
    this.setState({ isSubmitting: true });
    this.state.feeSet.map((item, index) => {
      if (!this.error) {
        for (const key in item) {
          // console.log(item[key] === "");
          if (item[key] === "" || item[key] === isNaN) {
            this.error = true;

            return this.setState({
              errorMessage: "This field cannot be empty.",
              errorKey: key,
              errorIndex: index,
              isSubmitting: false,
            });
          }
        }
        if (item.minAmount >= item.maxAmount) {
          this.error = true;
          this.setState({ isSubmitting: false });

          this.setState({
            errorMessage: `Amount must be greater than ${item.minAmount}.`,
            errorKey: "maxAmount",
            errorIndex: index,
          });
        }
      }
      if (item.minAmount <= this.maxAmount) {
        this.error = true;
        this.setState({ isSubmitting: false });

        this.setState({
          errorMessage: `Amount must be greater than ${this.maxAmount}.`,
          errorKey: "minAmount",
          errorIndex: index,
        });
      }

      this.maxAmount = item.maxAmount;

      return true;
    });
    if (!this.error) {
      let feeStructure = {
        paymentMethod: this.state.selectedPaymentPayout.paymentMethod,
        payoutMethod: this.state.selectedPaymentPayout.payoutMethod,
        currency: this.state.selectedCurrency.threeCharCode,
        sourceCountry: this.state.selectedSourceDestination.sourceThreeCharCode,
        destinationCountry: this.state.selectedSourceDestination
          .destThreeCharCode,
        feeRanges: this.state.feeSet,
      };
      if (this.state.edited) {
        let body = {
          id: this.props.location.data.feeSetId,
          feeStructure: feeStructure,
        };
        api
          .editFeeStructure(body)
          .then((res) => {
            this.setState({
              isSubmitting: false,
            });
            this.props.getFeeStructure();
            if (res.status === 400) {
              this.props.handleChange({
                showAlert: true,
                message: res.data.message,
                variant: "danger",
              });
            }
            else {
              this.props.handleChange({
                showAlert: true,
                message: "Fee set has been updated successfully.",
                variant: "success",
              });
            }
          })
          .catch((err) => {
            this.setState({
              isSubmitting: false,
            });
            this.props.handleChange({
              showAlert: true,
              message: "There was an error while updating fee set.",
              variant: "danger",
            });
          });
      } else {
        api
          .addFeeStructure(feeStructure)
          .then((res) => {
            this.setState({
              isSubmitting: false,
            });
            this.props.getFeeStructure();
            if (res.status === 400) {
              this.props.handleChange({
                showAlert: true,
                message: res.data.message,
                variant: "danger",
              });  
            }
            else {
              this.resetValue();
              this.props.handleChange({
                showAlert: true,
                message: "Fee set has been saved successfully.",
                variant: "success",
              });
            }
          })
          .catch((err) => {
            this.setState({
              isSubmitting: false,
            });
            this.props.handleChange({
              showAlert: true,
              message: "There was an error while saving fee set.",
              variant: "danger",
            });
          });
      }
    }
  };
  handleFeeSet = (values, index) => {
    let { name, value } = values;

    let feeSet = [...this.state.feeSet];
    feeSet[index] = {
      ...feeSet[index],
      [name]: value,
    };
    this.setState((prevState) => ({
      ...prevState,
      feeSet,
      errorMessage: "",
      errorKey: "",
      errorIndex: "",
    }));
  };
  addRow = () => {
    let item = { minAmount: "", maxAmount: "", flatFee: "", percentageFee: "" };
    this.setState({
      feeSet: [...this.state.feeSet, item],
    });
  };

  deleteRow = (deletedIndex) => {
    let filteredBank = this.state.feeSet.filter(
      (item, index) => index !== deletedIndex
    );

    this.setState({
      feeSet: filteredBank,
      errorMessage: "",
      errorKey: "",
      errorIndex: "",
    });
  };

  resetValue = () => {
    this.setState({
      selectedSourceDestination: {},
      selectedPaymentPayout: {},
      isSubmitting: false,
      selectedCurrency: {},
      feeSet: [
        { minAmount: "", maxAmount: "", flatFee: "", percentageFee: "" },
      ],
      edited: false,
      showAlert: false,
      message: "",
      variant: "",
      errorMessage: "",
      errorKey: "",
      errorIndex: "",
    });
  };
  render() {
    return (
      <>
        <Row>
          <Col
            style={{
              color: `${COLORS.BLACK_COLOR}`,
              fontWeight: "bold",
              fontSize: "22px",
            }}
          >
            {this.state.edited ? "Update Fee Set" : "Add Fee Set"}
            {/* Add Fee Set */}
          </Col>
        </Row>
        <Row className="mt-3">
          <Col>
            <Card className="p-3">
              <Row>
                <Col>
                  <Col
                    style={{
                      height: "50px",
                      backgroundColor: `${COLORS.LIGHT_BLUE}`,
                      border: ` 1px solid ${COLORS.BLUE_COLOR}`,
                    }}
                    className="d-flex align-items-center"
                  >
                    Select fee set parameter to set new fee range.{" "}
                  </Col>
                  <Row>
                    <Col>
                      <div className="mt-2 add-fee-structure">
                        <div className="add-fee-equal-w">
                          <label>Source destination</label>{" "}
                          <AdminDropdown
                            option={this.state.sourceDestination}
                            placeHolder="Select source destination"
                            pickerOption1="sourceName"
                            pickerOption2="destName"
                            onChangeOption={(value) => this.handleChange(value)}
                            customName="selectedSourceDestination"
                            customValue={this.state.selectedSourceDestination}
                          />
                        </div>
                        <div className="add-fee-equal-w add-fee-mid">
                          <label>Payment payout</label>{" "}
                          <AdminDropdown
                            option={this.state.paymentPayout}
                            placeHolder="Select source destination"
                            pickerOption1="paymentMethod"
                            pickerOption2="payoutMethod"
                            onChangeOption={(value) => this.handleChange(value)}
                            customName="selectedPaymentPayout"
                            customValue={this.state.selectedPaymentPayout}
                          />
                        </div>
                        <div className="add-fee-currency">
                          <label>Currency</label>
                          <Dropdown
                            option={this.state.currency}
                            placeHolder="Currency"
                            pickerOption="threeCharCode"
                            onChangeOption={(value) => this.handleChange(value)}
                            customName="selectedCurrency"
                            customValue={this.state.selectedCurrency}
                          />
                        </div>
                      </div>
                      {this.state.errorMessage && (
                        <div
                          style={{
                            color: "#dc3545",
                            backgroundColor: "rgba(255,0,0,.27)",
                            fontSize: "13px",
                            padding: "5px",
                            marginTop: "5px",
                          }}
                          className="d-flex justify-content-center"
                        >
                          {this.state.errorMessage}
                        </div>
                      )}

                      {!isEmpty(this.state.selectedSourceDestination) &&
                        !isEmpty(this.state.selectedPaymentPayout) &&
                        !isEmpty(this.state.selectedCurrency) && (
                          <>
                            <hr style={{ width: "100%", margin: "10px 0px" }} />

                            <Row className="rate-stat">
                              <Col>Min Amount</Col>
                              <Col>Max Amount</Col>
                              <Col>Flat Fee</Col>
                              <Col>Percentage Fee</Col>
                            </Row>
                            <Row>
                              <Col>
                                <hr
                                  style={{
                                    width: "100%",
                                    margin: "10px 0 15px 0",
                                  }}
                                />
                              </Col>
                            </Row>
                            {this.state.feeSet.map((item, index) => {
                              return (
                                <Row className="amount-field-wrap" key={index}>
                                  <Col className="amount-field">
                                    <InputField
                                      type="number"
                                      customWidth="100%"
                                      onChangeValue={(value) =>
                                        this.handleFeeSet(value, index)
                                      }
                                      customName="minAmount"
                                      customValue={item.minAmount}
                                      placeHolder={"0"}
                                      style={
                                        index === this.state.errorIndex &&
                                        "minAmount" === this.state.errorKey
                                          ? {
                                              border: "1px solid red",
                                            }
                                          : {}
                                      }
                                    />
                                  </Col>
                                  <Col className="amount-field">
                                    <InputField
                                      type="number"
                                      customWidth="100%"
                                      onChangeValue={(value) =>
                                        this.handleFeeSet(value, index)
                                      }
                                      customName="maxAmount"
                                      customValue={item.maxAmount}
                                      placeHolder={"0"}
                                      style={
                                        index === this.state.errorIndex &&
                                        "maxAmount" === this.state.errorKey
                                          ? {
                                              border: "1px solid red",
                                            }
                                          : {}
                                      }
                                    />
                                  </Col>
                                  <Col className="amount-field">
                                    <InputField
                                      type="number"
                                      customWidth="100%"
                                      onChangeValue={(value) =>
                                        this.handleFeeSet(value, index)
                                      }
                                      customName="flatFee"
                                      customValue={item.flatFee}
                                      placeHolder={"0"}
                                      style={
                                        index === this.state.errorIndex &&
                                        "flatFee" === this.state.errorKey
                                          ? {
                                              border: "1px solid red",
                                            }
                                          : {}
                                      }
                                    />
                                  </Col>
                                  <Col className="amount-field">
                                    <InputField
                                      type="number"
                                      customWidth={
                                        this.state.feeSet.length > 1
                                          ? "75%"
                                          : "100%"
                                      }
                                      onChangeValue={(value) =>
                                        this.handleFeeSet(value, index)
                                      }
                                      customName="percentageFee"
                                      customValue={item.percentageFee}
                                      placeHolder={"0"}
                                      style={
                                        index === this.state.errorIndex &&
                                        "percentageFee" === this.state.errorKey
                                          ? {
                                              border: "1px solid red",
                                            }
                                          : {}
                                      }
                                    />
                                    {this.state.feeSet.length > 1 && (
                                      <span className="ml-2">
                                        <button
                                          style={{
                                            backgroundColor: "transparent",
                                            border: "none",
                                          }}
                                          onClick={() => {
                                            this.deleteRow(index);
                                          }}
                                        >
                                          <img
                                            src={deleteImg}
                                            alt="deleteImg"
                                            height="20px"
                                            width="20px"
                                          />
                                        </button>
                                      </span>
                                    )}
                                  </Col>
                                </Row>
                              );
                            })}
                            <Row className="mt-4 mb-4">
                              <Col>
                                <Button
                                  variant="light"
                                  style={{
                                    border: `1px solid ${COLORS.GREY_COLOR}`,
                                  }}
                                  className="mr-3"
                                  onClick={() => this.addRow()}
                                >
                                  <img alt="icon" src={plusRound} />
                                </Button>
                                <ButtonSpinner
                                  block={false}
                                  isSubmitting={this.state.isSubmitting}
                                  title={" Save Fee Structure"}
                                  onClick={() => this.addFeeStructure()}
                                />
                                {this.state.edited && (
                                  <Button
                                    variant="danger"
                                    style={{
                                      border: `1px solid ${COLORS.WHITE_COLOR}`,
                                    }}
                                    className="ml-3"
                                    onClick={() => {
                                      this.props.getActiveAccountSideBar("FeeSet");                                      
                                      this.props.history.push(`${ROUTES.FEE_SET}`);
                                    }}
                                  >
                                    Cancel{" "}
                                  </Button>
                                )}
                              </Col>
                            </Row>
                          </>
                        )}
                    </Col>
                  </Row>
                </Col>
              </Row>
            </Card>
          </Col>{" "}
        </Row>
      </>
    );
  }
}
AddFeeSet.propTypes = {
  countries: PropTypes.array,
  location: PropTypes.object,
  handleChange: PropTypes.func,
  getFeeStructure: PropTypes.func,
  getActiveAccountSideBar: PropTypes.func,
  history: PropTypes.object,
};
const mapStateToProps = (state) => ({
  countries: state.miscellaneous.dest_country,
});
  
const mapDispatchToProps = { getFeeStructure, getActiveAccountSideBar };
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(AddFeeSet));
