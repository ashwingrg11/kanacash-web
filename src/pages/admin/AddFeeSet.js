import React, { Component } from "react";
import Header from "../../containers/layout/Header";
import { Container, Row, Col } from "react-bootstrap";
import Footer from "../../containers/layout/Footer";
import AdminSideBar from "../../containers/admin/AdminSideBar";
import AddFeeSetContain from "../../containers/admin/AddFeeSet";
import CustomAlert from "../../components/Alert/CustomAlert";
import Disclaimer from "../../components/Disclaimer/Disclaimer";
import { connect } from "react-redux";
class AddFeeSet extends Component {
  state = { showAlert: false, message: "", variant: "" };
  addEditFeeStructure = (value) => {
    this.setState(value);
  };
  render() {
    return (
      <>
        <Header />
        <div className="shadow-line"></div>
        <Container>
          <Row
            style={{
              minHeight: "900px",
            }}
          >
            <Col>
              <Row>
                <Col xl="3" lg="3" md="3" sm="12" style={{ marginTop: "40px" }}>
                  <AdminSideBar />
                </Col>
                <Col xl="9" lg="9" md="9" sm="12" style={{ marginTop: "40px" }}>
                  <AddFeeSetContain
                    handleChange={(value) => this.addEditFeeStructure(value)}
                  />
                </Col>
              </Row>
            </Col>
          </Row>
        </Container>

        <Footer />
        {!localStorage.acceptPolicy && <Disclaimer />}

        <CustomAlert
          show={this.state.showAlert}
          variant={this.state.variant}
          message={this.state.message}
          close={() => this.setState({ showAlert: false })}
        />
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  acceptPolicy: state.auth.accept_policy,
});

export default connect(mapStateToProps)(AddFeeSet);
