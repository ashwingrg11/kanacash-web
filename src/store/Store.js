import { createStore, combineReducers, applyMiddleware, compose } from "redux";
// import { composeWithDevTools } from 'redux-devtools-extension';
import Thunk from "redux-thunk";
import LoginReducer from "./reducers/Login";
import ForgotPasswordReducer from "./reducers/ForgotPassword";
import LoaderReducer from "./reducers/Loader";
import ModalReducer from "./reducers/Modal";
import NetworkReducer from "./reducers/Network";
import AuthReducer from "./reducers/Auth";
import ErrorReducer from "./reducers/Error";
import MiscellaneousReducer from "./reducers/MiscellaneousReducer";
import senderDetailReducers from "./reducers/SenderDetailReducer";
import TransactionReducer from "./reducers/TransactionReducer";
import SignUpReducer from "./reducers/SignUpReducer";
import AdminReducers from "./reducers/adminReducers";

const middleware = [Thunk];
const initialState = {};

const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
      })
    : compose;

const rootReducer = combineReducers({
  login: LoginReducer,
  forgotPassword: ForgotPasswordReducer,
  loader: LoaderReducer,
  modal: ModalReducer,
  network: NetworkReducer,
  auth: AuthReducer,
  error: ErrorReducer,
  miscellaneous: MiscellaneousReducer,
  senderDetail: senderDetailReducers,
  transaction: TransactionReducer,
  signup: SignUpReducer,
  admin: AdminReducers,
});

const store = createStore(
  rootReducer,
  initialState,
  composeEnhancers(applyMiddleware(...middleware))
);
export default store;
