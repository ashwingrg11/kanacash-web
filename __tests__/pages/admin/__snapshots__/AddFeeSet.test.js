import React from "react";
import AddFeeSet from "../../../../src/pages/admin/AddFeeSet";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";
import store from "../../../../src/store/Store";

import renderer from "react-test-renderer";
describe("<AddFeeSet />", () => {
  it("AddFeeSet renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <AddFeeSet />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
