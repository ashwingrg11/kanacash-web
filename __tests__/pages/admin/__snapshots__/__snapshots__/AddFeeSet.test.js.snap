// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddFeeSet /> AddFeeSet renders correctly 1`] = `
Array [
  <header
    className="header-main"
  >
    <div
      className="container"
    >
      <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          className="navbar-brand"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <img
            alt="logo"
            className="logo"
            src=""
          />
        </div>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            className="ml-md-auto navbar-nav"
            onKeyDown={[Function]}
          >
            <span
              className="btn btn-primary btn-theme btn-primary-border"
              onClick={[Function]}
            >
              Sign in
            </span>
            <button
              className="btn btn-primary btn-theme btn-primary-theme"
              onClick={[Function]}
            >
              Sign up
            </button>
          </div>
        </div>
      </nav>
    </div>
  </header>,
  <div
    className="shadow-line"
  />,
  <div
    className="container"
  >
    <div
      className="row"
      style={
        Object {
          "minHeight": "900px",
        }
      }
    >
      <div
        className="col"
      >
        <div
          className="row"
        >
          <div
            className="col-xl-3 col-lg-3 col-md-3 col-sm-12"
            style={
              Object {
                "marginTop": "40px",
              }
            }
          >
            <div
              className="row"
              style={
                Object {
                  "backgroundColor": "#FAFAFA",
                  "padding": "20px",
                }
              }
            >
              <div
                className="col"
              >
                <div
                  className="mt-2 row"
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                >
                  <div
                    className="d-flex align-items-center col"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Dashboard
                  </div>
                </div>
                <div
                  className="mt-2 row"
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                >
                  <div
                    className="d-flex align-items-center col"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Add Fee Set
                  </div>
                </div>
                <div
                  className="mt-2 row"
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                >
                  <div
                    className="d-flex align-items-center col"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Fee Set
                  </div>
                </div>
                <div
                  className="mt-2 row"
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                >
                  <div
                    className="d-flex align-items-center col"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Unlock Sender
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-xl-9 col-lg-9 col-md-9 col-sm-12"
            style={
              Object {
                "marginTop": "40px",
              }
            }
          >
            <div
              className="row"
            >
              <div
                className="col"
                style={
                  Object {
                    "color": "#212529",
                    "fontSize": "22px",
                    "fontWeight": "bold",
                  }
                }
              >
                Add Fee Set
              </div>
            </div>
            <div
              className="mt-3 row"
            >
              <div
                className="col"
              >
                <div
                  className="p-3 card"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col"
                    >
                      <div
                        className="d-flex align-items-center col"
                        style={
                          Object {
                            "backgroundColor": "#d1ecf1",
                            "border": " 1px solid rgba(16, 119, 190, 0.1)",
                            "height": "50px",
                          }
                        }
                      >
                        Select fee set parameter to set new fee range.
                         
                      </div>
                      <div
                        className="row"
                      >
                        <div
                          className="col"
                        >
                          <div
                            className="mt-2 add-fee-structure"
                          >
                            <div
                              className="add-fee-equal-w"
                            >
                              <label>
                                Source destination
                              </label>
                               
                              <div
                                className="sc-AxjAm jhiTqR"
                              >
                                <div
                                  className="sc-AxirZ fpkyRk input-dropdown"
                                  style={
                                    Object {
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <div
                                    className="sc-AxhCb flAYhK custom-wrap"
                                    onClick={[Function]}
                                  >
                                    <div
                                      className="sc-AxgMl fqhLTf"
                                    >
                                      <div
                                        className="selected-country"
                                      >
                                        <span>
                                          Select source destination
                                        </span>
                                      </div>
                                      <img
                                        alt="arrow-dwon"
                                        src=""
                                      />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                            <div
                              className="add-fee-equal-w add-fee-mid"
                            >
                              <label>
                                Payment payout
                              </label>
                               
                              <div
                                className="sc-AxjAm jhiTqR"
                              >
                                <div
                                  className="sc-AxirZ fpkyRk input-dropdown"
                                  style={
                                    Object {
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <div
                                    className="sc-AxhCb flAYhK custom-wrap"
                                    onClick={[Function]}
                                  >
                                    <div
                                      className="sc-AxgMl fqhLTf"
                                    >
                                      <div
                                        className="selected-country"
                                      >
                                        <span>
                                          Select source destination
                                        </span>
                                      </div>
                                      <img
                                        alt="arrow-dwon"
                                        src=""
                                      />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                            <div
                              className="add-fee-currency"
                            >
                              <label>
                                Currency
                              </label>
                              <div
                                className="sc-AxjAm jhiTqR"
                              >
                                <div
                                  className="sc-AxirZ fpkyRk input-dropdown"
                                  style={
                                    Object {
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <div
                                    className="sc-AxhCb flAYhK custom-wrap"
                                    onClick={[Function]}
                                  >
                                    <div
                                      className="sc-AxgMl fqhLTf"
                                    >
                                      <div
                                        className="selected-country"
                                      >
                                        <span>
                                          Currency
                                        </span>
                                      </div>
                                      <img
                                        alt="arrow-dwon"
                                        src=""
                                      />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                          
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="footer-section text-center"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="foot-nav"
          >
            <a
              aria-current="page"
              className="link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
            <a
              aria-current={null}
              className="link"
              href="/about-us"
              onClick={[Function]}
            >
              About Us
            </a>
            <a
              aria-current={null}
              className="link"
              href="/terms-of-service"
              onClick={[Function]}
            >
              Terms
            </a>
            <a
              aria-current={null}
              className="link"
              href="/privacy-policy"
              onClick={[Function]}
            >
              Privacy Policy
            </a>
            <a
              aria-current={null}
              className="link"
              href="/copyright"
              onClick={[Function]}
            >
              Copyright
            </a>
            <a
              aria-current={null}
              className="link"
              href="/contact-us"
              onClick={[Function]}
            >
              Contact
            </a>
          </div>
        </div>
        <div
          className="copyright col-sm-12"
        >
          <p>
            © Golden Money Transfer Inc. 2020. All Rights Reserved.
             
            <b>
              Licensed by
            </b>
             Department of Financial Institutions, Nevada
          </p>
        </div>
      </div>
    </div>
  </footer>,
  <div
    className="disclaimer text-center"
  >
    <div
      className="container"
    >
      <div
        className="policy"
      >
         
        This website uses cookies to ensure you get the best experience on our website, as described in our
         
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            privacy policy.
        </span>
      </div>
      <div
        className="policy1"
      >
        This website uses cookies,
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            learn more.
        </span>
      </div>
      <div
        className="buttonDiv"
      >
        <button
          className="btn-outline-warning"
          onClick={[Function]}
          style={
            Object {
              "border": "none",
              "borderRadius": "50px",
              "color": "black",
              "padding": "5px 15px",
            }
          }
        >
          Accept
        </button>
      </div>
    </div>
  </div>,
]
`;
